src/coprocessor/statistics/analyze.rs

```diff
@@ -77,23 +81,44 @@ impl<S: Snapshot> AnalyzeContext<S> {
     // it would build a histogram and count-min sketch of index values.
     fn handle_index(
         req: AnalyzeIndexReq,
-        scanner: &mut IndexScanExecutor<TiKVStorage<SnapshotStore<S>>>,
+        scanner: &mut RangesScanner<TiKVStorage<SnapshotStore<S>>>,
+        is_common_handle: bool,
     ) -> Result<Vec<u8>> {
         let mut hist = Histogram::new(req.get_bucket_size() as usize);
         let mut cms = CmSketch::new(
             req.get_cmsketch_depth() as usize,
             req.get_cmsketch_width() as usize,
         );
-        while let Some(row) = scanner.next()? {
-            let row = row.take_origin()?;
-            let (bytes, end_offsets) = row.data.get_column_values_and_end_offsets();
-            hist.append(bytes);
-            if let Some(c) = cms.as_mut() {
-                for end_offset in end_offsets {
-                    c.insert(&bytes[..end_offset])
+
+        while let Some((key, _)) = scanner.next()? {
+            let mut key = &key[..];
+            if is_common_handle {
+                table::check_record_key(key)?;
+                key = &key[table::PREFIX_LEN..];
+            } else {
+                table::check_index_key(key)?;
+                key = &key[table::PREFIX_LEN + table::ID_LEN..];
+            }
+            let mut datums = key;
+            let mut data = Vec::with_capacity(8 * req.get_num_columns() as usize);
```

Comment:
What does 8 comes from? What if the column is string longer than 8 bytes?

Reply:
Done.